woocommerce/woocommerce-gutenberg-products-block
# [Store Customization] Make request to the Verticals API to fetch matching images and classify them

## What

Adds the `PatternImages` class.

Fixes #10620

## Why

We need to get the images for a given vertical from the Verticals API, and then assign them to each of the patterns that need images in the store.

## Testing Instructions



Automated tests are in `tests/php/Patterns/PatternImagesTest.php`.

1. Add this code somewhere in the plugin (for example on the `woocommerce-gutenberg-products-block.php`)
```php
use Automattic\WooCommerce\Blocks\Patterns\PatternImages;
use Automattic\WooCommerce\Blocks\Verticals\Client;

$dictionary = wp_json_file_decode( __DIR__ . '/src/Patterns/dictionary.json', array( 'associative' => true ) );

$pi = new PatternImages(
	new Client(),
	$dictionary
);
$pi->create_patterns_content( 34 );

$patterns_content = get_option( 'wc_blocks_patterns_content' );
var_dump( $patterns_content );
```
2. Go to the home page and check the `wc_blocks_patterns_content` option has stored data.


## Screenshots or screencast

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog
